conduit/pipeline/pipeline_test.go

```diff
@@ -866,3 +867,114 @@ func TestRoundOverwrite(t *testing.T) {
 		assert.Equal(t, uint64(i), pImpl.pipelineMetadata.NextRound)
 	}
 }
+
+// an importer that simply errors out when GetBlock() is called
+type errorImporter struct {
+	genesis *bookkeeping.Genesis
+}
+
+var errorImporterMetadata = conduit.Metadata{
+	Name:         "error_importer",
+	Description:  "An importer than errors out whenever GetBlock() is called",
+	Deprecated:   false,
+	SampleConfig: "",
+}
+
+// New initializes an importer
+func New() importers.Importer {
+	return &errorImporter{}
+}
+
+func (e errorImporter) Metadata() conduit.Metadata {
+	return errorImporterMetadata
+}
+
+func (e errorImporter) Init(_ context.Context, _ plugins.PluginConfig, _ *log.Logger) (*bookkeeping.Genesis, error) {
+	return e.genesis, nil
+}
+
+func (e errorImporter) Config() string {
+	return ""
+}
+
+func (e errorImporter) Close() error {
+	return nil
+}
+
+func (e errorImporter) GetBlock(_ uint64) (data.BlockData, error) {
+	return data.BlockData{}, fmt.Errorf("error maker")
+}
+
+// TestPipelineRetryVariables tests that modifying the retry variables results in longer time taken for a pipeline to run
+func TestPipelineRetryVariables(t *testing.T) {
+	tests := []struct {
+		name             string
+		retryDelayString string
+		retryCount       uint64
+		totalDuration    time.Duration
+		epsilon          time.Duration
+	}{
+		{"2 seconds", "2s", 1, 2 * time.Second, 1 * time.Second},
+		{"10 seconds", "2s", 5, 10 * time.Second, 1 * time.Second},
+	}
+	for _, testCase := range tests {
+		t.Run(testCase.name, func(t *testing.T) {
+
+			var pImporter importers.Importer = &errorImporter{genesis: &bookkeeping.Genesis{Network: "test"}}
```

Comment:
With testify mocks you can also assert that GetBlock is called "retryCount + 1" times. We're using it in one of the other tests:
https://github.com/algorand/indexer/blob/develop/conduit/pipeline/pipeline_test.go#L286